src/generator.cs

```diff
@@ -7078,6 +7081,9 @@ bool ShouldSkipGeneration (MethodInfo mi)
 		if (mi.IsUnavailable ())
 			return true;
 
+		// Skip those methods marked to be ignored by the developer
+		if (mi.GetCustomAttributes (true).OfType<IgnoredInDelegateAttribute> ().Any ())
```

Comment:
`mi.GetCustomAttributes (true)` is slow (reflection) and allocates memory. For classic (which XM will keep around longer than XI) there's an identical call just after. Please merge both :-)